src/assets/php8.4/php8.4.md

```diff
@@ -0,0 +1,234 @@
+  4. Implicitly nullable parameter declarations depreciated
+  5. New multibyte functions
+
+## 1. New array helper functions
+
+The following variants of array helper functions will be added in PHP 8.4:
+
+  * `array_find()`
+  * `array_find_key()`
+  * `array_any()`
+  * `array_all()`
+
+These functions will take an array and a callback function and return the following:
+
+
+| functions | Return value |
+| :---- | :---- |
+| `array_find()` | Returns the first element that meets the callback condition; `NULL` otherwise. |
+| `array_find_key()` | Returns the key of the first element that meets the callback condition; `NULL` otherwise. |
+| `array_any()` | Returns `true` if at least one element matches the callback condition; `false` otherwise. |
+| `array_all()` | Returns `true` if all elements match the callback condition; `false` otherwise. |
+
+ Note: `array_find()` retrieves only the first matching element. For multiple matches, consider using `array_filter()`.
+
+### Example
+
+Given an array with key-value pairs and a callback function:
+
+```
+$array = ['1'=> 'red', '2'=> 'purple', '3' => 'green']
+
+function hasLongName($value) {
+  return strlen($value) > 4;
+}
+```
+
+Here's how we can use the new functions:
+
+1. `array_find()`:
+
+```
+// Find the first color with a name length greater than 4
+
+$result1 = array_find($array, 'hasLongName');
+
+var_dump($result1);  // string(5) "purple"
+```
+
+2. `array_find_key()`:
+
+```
+// Find the key of the first color with a name length greater than 4
+
+$result2 = array_find_key($array, 'hasLongName');
+
+var_dump($result2);  // string(1) "2"
+```
+
+3. `array_any()`:
+
+```
+// Check if any color name has a length greater than 4
+
+$result3 = array_any($array, 'hasLongName');
+
+var_dump($result3);  // bool(true)
+```
+
+4. `array_all()`:
+
+```
+// Check if all color names have a length greater than 4
+
+$result4 = array_all($array, 'hasLongName');
+
+var_dump($result4);  // bool(false)
+```
+
+## 2. Property hooks
+
+PHP 8.4 introduces property hooks, offering a more elegant way to access and modify private or protected properties of a class. Previously, developers relied on getters, setters, and magic methods (`__get` and `__set`). Now, you can define `get` and `set` hooks directly on a property, reducing the boilerplate code.
+
+Instead of ending the property with a semicolon, we can use a code block `{}` to include the property hook.
+These hooks are optional and can be used independently. By excluding one or the other we can make the property read-only or write-only.
+
+### Example
+
+```
+class User
+{
+    public function __construct(private string $first, private string $last) {}
+
+    public string $fullName {
+      get => $this->first . " " . $this->last;
+
+      set ($value) {
+        if (!is_string($value)) {
+          throw new InvalidArgumentException("Expected a string for full name, " . gettype($value) . " given.");
+        }
+	    if (strlen($value) === 0) {
```

Comment:
The indent here is a bit different

Reply:
I couldn't figure out this indentation problem:
https://github.com/JankariTech/blog/pull/157#discussion_r1749660381

Reply:
figured out now